### ISSUE TITLE
Unused arguments to an async function don't show up in the debugger

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/sdk#28177]
Look at the bottom right window in each case.

Here, you see all three arguments 
![sync_see_all_three](https://cloud.githubusercontent.com/assets/17034/21401991/572f3452-c76b-11e6-9487-decc86b21bae.png)

Here, you only see the two used args
![async_missing_c](https://cloud.githubusercontent.com/assets/17034/21401990/572c72ee-c76b-11e6-8b70-d451169266c0.png)

Good find, @Pacane

**Comment by johnmccutchan** [comment]
This is probably working as intended because the asynchronous body is actually a closure and we aren't capturing unused parameters

**Comment by johnmccutchan** [comment]
@mhausner

**Comment by Pacane** [comment]
It's quite unexpected when you're unaware of that implementation detail though.

**Comment by johnmccutchan** [comment]
@Pacane Agreed it can be surprising. We could probably adjust the implementation so that unused parameters are always captured, but, if the code doesn't ever reference a variable why would you need to inspect it in the debugger? It cannot affect the outcome of the code being executed.

**Comment by kevmoo** [comment]
@johnmccutchan I think it's just about dev experience – not so much when debugging a working app, but when you're getting started. Understanding what you're getting passed in from the framework, etc.

**Comment by Pacane** [comment]
That's correct, I practice what I call DDD (debugger driven development).

As a serious note, it's pretty much what @kevmoo said. When you try a new
lib/framework, it's likely that you don't know what's the API, and what's
easier than   try/run it to figure out.

**Comment by kevmoo** [comment]
Even listing out the unused variable as `In an async function, unused argument values are not visible` – just so folks like @Pacane don't think they are going crazy

**Comment by zoechi** [comment]
I'm sure I reported this a few years ago already. 
When you investigate the current status at a breakpoint then the values in scope **are** relevant and helpful even when they are not used in a closure and usually help to understand why an application reached the current status.

Also not showing getters is really painful when investigating the status.

**Comment by mhausner** [comment]
@johnmccutchan  is right. The variables are not showing up because they are not captured. You don't want them to be captured because you don't want to pay the overhead of captured variables unless it's necessary. As @zoechi says, this has been reported before. However, it does not make sense to pay the runtime overhead to maintain info just for the debugger.

**Comment by Pacane** [comment]
Could the debugger know what variables were captured or not? If so, it'd be
great to have some sort of message saying the variable can't be inspected
since it's not used/captured.
